src/main/java/nus/climods/ui/module/ModuleCard.java

```diff
@@ -45,15 +45,22 @@ public class ModuleCard extends UiPart<Region> {
     /**
      * Creates a {@code PersonCode} with the given {@code Person} and index to display.
      */
-    public ModuleCard(DummyModule module) {
+    public ModuleCard(Module module) {
         super(FXML);
         this.module = module;
-        moduleCode.setText(module.getModuleCode());
+        moduleCode.setText(module.getCode());
         title.setText(module.getTitle());
         department.setText(module.getDepartment());
         module.getSemesterData()
-            .forEach(tag -> moduleInfo.getChildren().add(new Pill(tag, SEMESTER_BG_COLOR, SEMESTER_TEXT_COLOR, 13)));
-        moduleInfo.getChildren().add(new Pill(module.getWorkload(), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR, 13));
+            .forEach(semData -> {
+                assert semData.getSemester() != null;
+                moduleInfo.getChildren()
+                    .add(new Pill(String.format("Semester %s", semData.getSemester().toString()), SEMESTER_BG_COLOR,
+                        SEMESTER_TEXT_COLOR, 13));
+            });
+        moduleInfo.getChildren()
+            .add(new Pill(String.format("%s MCs", module.getModuleCredit()), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR,
```

Comment:
If we change `getModuleCredit()` to return a double, maybe we could still keep truncation of decimal places for integer MCs like this? Alternatively this code could be moved into the Module class itself to another getter method.
```java
Double moduleCredit = module.getModuleCredit();
String formattedModuleCredit = Math.ceil(moduleCredit) == Math.floor(moduleCredit)
        ? Integer.toString(moduleCredit.intValue()) : Double.toString(moduleCredit);

moduleInfo.getChildren()
    .add(new Pill(String.format("%s MCs", formattedModuleCredit), WORKLOAD_BG_COLOR, WORKLOAD_TEXT_COLOR,
                13));
```